lib/Sema/TypeCheckDistributed.cpp

```diff
@@ -434,11 +433,17 @@ static bool checkDistributedTargetResultType(
 
 /// Check whether the function is a proper distributed function
 ///
-/// \param diagnose Whether to emit a diagnostic when a problem is encountered.
-///
 /// \returns \c true if there was a problem with adding the attribute, \c false
 /// otherwise.
-bool swift::checkDistributedFunction(FuncDecl *func, bool diagnose) {
+bool swift::checkDistributedFunction(AbstractFunctionDecl *func) {
+  auto &C = func->getASTContext();
+  return evaluateOrDefault(C.evaluator,
+                           CheckDistributedFunctionRequest{func},
+                           true);
```

Comment:
I think if there was a cycle in the graph the default should be false.

Reply:
Okey, let's do that

Reply:
done